src/logic/function/basic.lean

```diff
@@ -540,34 +540,53 @@ along a function `f : α → β` to a function `β → γ`,
 by using the values of `g` on the range of `f`
 and the values of an auxiliary function `e' : β → γ` elsewhere.
 
-Mostly useful when `f` is injective. -/
+Mostly useful when `f` is injective, more generally when `g.factors_through f`. -/
 def extend (f : α → β) (g : α → γ) (e' : β → γ) : β → γ :=
 λ b, if h : ∃ a, f a = b then g (classical.some h) else e' b
 
+/-- g factors through f : `f a = f b → g a = g b` -/
+def factors_through (g : α → γ) (f : α → β) : Prop :=
+∀ ⦃a b⦄, f a = f b → g a = g b
+
+lemma factors_through_of_is_injective (g : α → γ) (hf : injective f) :
+g.factors_through f :=  λ a b h, congr_arg g (hf h)
```

Comment:
```suggestion
lemma injective.factors_through (g : α → γ) (hf : injective f) : g.factors_through f :=
λ a b h, congr_arg g (hf h)
```

Reply:
I see, this makes `hf.factors_through` work for `hf : f.injective`?